crates/chain-state/src/in_memory.rs

```diff
@@ -798,28 +798,83 @@ impl<N: NodePrimitives> ExecutedBlock<N> {
     }
 }
 
+/// Trie updates that result from calculating the state root for the block.
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum ExecutedTrieUpdates {
+    /// Trie updates present. State root was calculated, and the trie updates can be applied to the
+    /// database.
+    Present(Arc<TrieUpdates>),
+    /// Trie updates missing. State root was calculated, but the trie updates cannot be applied to
+    /// the current database state. To apply the updates, the state root must be recalculated, and
+    /// new trie updates must be generated.
+    ///
+    /// This can happen when processing fork chain blocks that are building on top of the
+    /// historical database state. Since we don't store the historical trie state, we cannot
+    /// generate the trie updates for it.
+    Missing,
+}
+
+impl ExecutedTrieUpdates {
+    /// Creates a [`ExecutedTrieUpdates`] with present but empty trie updates.
+    pub fn empty() -> Self {
```

Comment:
why do we need this method? if only for testing, let's hide it from public API with `cfg(test)`

Reply:
oh wait nope, it's used in ress
```
error[E0599]: no variant or associated item named `empty` found for enum `ExecutedTrieUpdates` in the current scope
   --> /home/runner/work/reth/reth/crates/ress/provider/src/lib.rs:133:60
    |
133 | ...                   trie: ExecutedTrieUpdates::empty(),
    |                                                  ^^^^^ variant or associated item not found in `ExecutedTrieUpdates`
```